ext/otel_observer.c

```diff
@@ -800,7 +968,39 @@ static otel_observer *resolve_observer(zend_execute_data *execute_data) {
 
     if (!zend_llist_count(&observer_instance.pre_hooks) &&
         !zend_llist_count(&observer_instance.post_hooks)) {
-        return NULL;
+        if (OTEL_G(attr_hooks_enabled) && has_withspan_attribute) {
```

Comment:
Perhaps there should be some log message somewhere to at least let people know that `WithSpan` was ignored? Also maybe we should discuss potentially allowing both in the future, but it's definitely safer to initially go with a solution where this is not possible.

Reply:
I'll add a log message... not sure what you mean by "allowing both" ?

Reply:
added a warning message